tools/gstf_preparation/gstf_preparation.py

```diff
@@ -449,17 +465,19 @@ def __main__():
                     if transcript_biotype and transcript_biotype != 'protein_coding':
                         print("Transcript %s has biotype %s (not protein-coding), removing from FASTA output" % (transcript_id, transcript_biotype), file=sys.stderr)
                         continue
+                    if transcript_id not in selected_transcript_ids:
+                        continue
 
                 if options.headers == "TranscriptId_species":
                     # Change the FASTA header to '>TranscriptId_species', as required by TreeBest
                     # Remove any underscore in the species
                     entry.header = ">%s_%s" % (transcript_id, transcript['species'].replace('_', ''))
                 elif options.headers == "GeneSymbol-TranscriptID_species":
```

Comment:
Forgot to update here, but I'm too kind and will do it :stuck_out_tongue_closed_eyes: